nameserver/server.go

```diff
@@ -54,51 +65,64 @@ func (proto dnsProtocol) String() string {
 }
 
 // get a new dns.Client for a protocol
-func (proto dnsProtocol) GetNewClient(bufsize int) *dns.Client {
+func (proto dnsProtocol) GetNewClient(bufsize int, timeout time.Duration) *dns.Client {
 	switch proto {
 	case protTCP:
-		return &dns.Client{Net: "tcp"}
+		return &dns.Client{Net: "tcp", ReadTimeout: timeout}
 	case protUDP:
-		return &dns.Client{Net: "udp", UDPSize: uint16(bufsize)}
+		return &dns.Client{Net: "udp", ReadTimeout: timeout, UDPSize: uint16(bufsize)}
 	}
 	return nil
 }
 
```

Comment:
`server.go` has changed so much that I doubt it would be worth to review it by using the diff... The same applies to `zone.go`